Wikipedia/Code/PageHistoryViewController.m

```diff
@@ -191,4 +194,23 @@ - (void)didReceiveMemoryWarning {
     // Dispose of any resources that can be recreated.
 }
 
+- (BOOL)shouldLoadNewData {
+    
+    CGFloat maxY = self.tableView.contentOffset.y + self.tableView.frame.size.height;
+    BOOL shouldLoad = NO;
+    if (!self.loadInProgress && maxY >= self.tableView.contentSize.height) {
```

Comment:
Nitpick: consider moving the `loadInProgress` check to the first line of the method so it can bail out before calculating `maxY`.